meta/CSVExporter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace dokuwiki\plugin\struct\meta;
+
+/**
+ * Class CSVExporter
+ *
+ * exports raw schema data to CSV. For lookup schemas this data can be reimported again through
+ * CSVImporter
+ *
+ * Note this is different from syntax/csv.php
+ *
+ * @package dokuwiki\plugin\struct\meta
+ */
+class CSVExporter {
+
+    protected $islookup = false;
+
+    /**
+     * CSVImporter constructor.
+     *
+     * @throws StructException
+     * @param string $table
+     */
+    public function __construct($table) {
+
+        $search = new Search();
+        $search->addSchema($table);
+        $search->addColumn('*');
+        $result = $search->execute();
+
+        $this->islookup = $search->getSchemas()[0]->isLookup();
+        if(!$this->islookup) {
+            $pids = $search->getPids();
+        } else {
+            $pids = array();
+        }
+
+        echo $this->header($search->getColumns());
+        foreach($result as $i => $row) {
+            if(!$this->islookup) {
+                $pid = $pids[$i];
+            } else {
+                $pid = 0;
+            }
+
+            echo $this->row($row, $pid);
+        }
+    }
+
+    /**
+     * Create the header
+     *
+     * @param Column[] $columns
+     * @return string
+     */
+    protected function header($columns) {
+        $row = '';
+
+        if(!$this->islookup) {
+            $row .= $this->escape('pid');
```

Comment:
I believe this is what Excel and LibreOffice do by default, too. CSV is a confusing file format anyway. Keeping all fields enclosed in quotes makes the file a little bit more consistent and easier to parse for 3rd party software.